lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -36,12 +34,15 @@ def execute(self):
             # We were not given any remotes on the command line, build the list
             # from the tools group directory.
             remotes = self.remote(self.tool_group_dir)
+            if not remotes:
+                self.logger.error(f'No such group "{self.context.group}".')
+                return 1
 
         for remote in remotes:
             tg_dir_r = self.tool_group_dir / remote
             if not tg_dir_r.exists():
                 self.logger.warn(
-                    'The given remote host, "%s", is not a directory in' " %s.",
+                    'The given remote host, "%s", is not a directory in %s.',
```

Comment:
Normally as a "log" message I'd say this is OK, but since these seem to be the only user output rather than internal logs, complaining about the internal representation of "remote" as a subdirectory seems wonky. Shouldn't we say something more like "The remote host isn't in tool group 'group'"?

Reply:
Probably.  My intention with this change was just to remove the string concatenation (and to make the quotation marks uniform).  Of course, in this context, the arguments actually are directory names/paths, so the message makes sense...although, I concur, this doesn't necessarily help the end-user much.

Reading over these two scripts, I was struck by how not-particularly-user-friendly it is.  But, I didn't really want to scrap it completely, so I tried to touch up only the most egregious bits and pass over the rest.

I really wish there were a "style guide" for diagnostic messages.  I feel like we had something like that back in the VMS days...and it kinda got lost with the move to Unix and now it's completely gone on Linux....